tests/unit/resources/test_virtual_machines.py

```diff
@@ -285,6 +285,18 @@ def test_power_off(self, mock_get, mock_post):
         vm.power_off()
         mock_post.assert_called_once_with('/virtual_machines/12345/power_off', None, custom_headers={'Content-type': 'application/vnd.simplivity.v1.11+json'})
 
+    @mock.patch.object(Connection, "post")
+    @mock.patch.object(Connection, "get")
+    def test_power_on(self, mock_get, mock_post):
+        mock_post.return_value = None, [{'object_id': '12345'}]
+        mock_get.return_value = {'virtual_machines': {'id': '12345'}}
+
+        vm1_data = {'name': 'name1', 'id': '12345'}
+        vm = self.machines.get_by_data(vm1_data)
+
+        vm.power_on()
+        mock_post.assert_called_once_with('/virtual_machines/12345/power_on', None, custom_headers={'Content-type': 'application/vnd.simplivity.v1.14+json'})
```

Comment:
check the data within vm to ensure the refresh worked

Reply:
Done